src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,10 @@
 
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
```

Comment:
Use bound for T - we should not be able to have MachineProducer<String>
![image](https://user-images.githubusercontent.com/20326858/202157015-cd7f27ed-e8a5-4c6b-a49b-07ca230cde7a.png)